client/signup/config/flows.js

```diff
@@ -328,16 +335,15 @@ const Flows = {
 		}
 
 		/**
-		 * This is called on Signup start (page initialization),
-		 * before the steps are rendered. Used if there is a need
-		 * to filter the first step in the flow.
+		 * If there is need to test the first step in a flow,
+		 * the best way to do it is to check for:
+		 *
+		 * 	if ( '' === stepName ) { ... }
+		 *
+		 * This will be fired at the beginning of the signup flow.
 		 */
-		if ( '' === stepName ) {
-			abtest( 'noSurveyStep' );
-		}
-
 		if ( 'main' === flowName ) {
-			if ( 'survey' === stepName ) {
+			if ( '' === stepName ) {
```

Comment:
This is needed, because the way the Signup framework works right now, we should generate the AB variation at the very start of signup, where `getFlow` is called without a `currentStep` parameter. 

Additionally `getFlow` is called quite a lot of times on every step change without the step name, which doesn't affect that, except that it's called numerous extra times. 

`abtest` has a check to return the already set value, so even if we add a check to avoid multiple calls, the benefit will be minuscule.